File: Context.swift
/* CONTEXT */

class Context {
    var display_name: String 
    var parent: Context?
    var parent_entry_pos: Position?

    init(display_name: String, parent: Context?=nil, parent_entry_pos: Position?=nil) {
        self.display_name = display_name
        self.parent = parent
        self.parent_entry_pos = parent_entry_pos
    }

    init() {
        self.display_name = ""
        self.parent = nil 
        self.parent_entry_pos = nil 
    }
}

File: Context.py
from SymbolTable import SymbolTable 

class Context:
    def __init__(self, display_name=None, parent=None, parent_entry_pos=None):
        self.display_name = display_name
        self.parent = parent 
        self.parent_entry_pos = parent_entry_pos
        self.symbolTable = SymbolTable()
